llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -398,15 +398,14 @@ void SelectionDAGISelLegacy::getAnalysisUsage(AnalysisUsage &AU) const {
   AU.addRequired<TargetLibraryInfoWrapperPass>();
   AU.addRequired<TargetTransformInfoWrapperPass>();
   AU.addRequired<AssumptionCacheTracker>();
-  if (UseMBPI && OptLevel != CodeGenOptLevel::None)
-      AU.addRequired<BranchProbabilityInfoWrapperPass>();
+  if (UseMBPI)
```

Comment:
I think we can add extra check for PGO (`Selector->TM.getPGOOption()`) and only do the analysis when PGO is used.